packages/compass-sidebar/src/components/multiple-connections/sidebar.tsx

```diff
@@ -240,20 +245,32 @@ export function MultipleConnectionSidebar({
               }
             }}
             initialConnectionInfo={editingConnectionInfo}
-            onSaveAndConnectClicked={(connectionInfo) => {
-              void connect(connectionInfo);
-            }}
-            onSaveClicked={(connectionInfo) => {
-              return saveEditedConnection(connectionInfo);
-            }}
-            onCancel={() => {
-              cancelEditConnection(editingConnectionInfo.id);
-            }}
             connectionErrorMessage={
               connectionErrors[editingConnectionInfo.id]?.message
             }
             openSettingsModal={openSettingsModal}
             {...formPreferences}
+            onCancel={() => {
+              cancelEditConnection(editingConnectionInfo.id);
+            }}
+            onSaveClicked={(connectionInfo) => {
+              return saveEditedConnection(connectionInfo);
+            }}
+            onConnectClicked={
+              isEditingNewConnection || disableEditingConnectedConnection
+                ? undefined
+                : (connectionInfo) => {
+                    void connect(connectionInfo);
+                  }
+            }
+            onSaveAndConnectClicked={
+              disableEditingConnectedConnection
+                ? undefined
+                : (connectionInfo) => {
+                    void saveEditedConnection(connectionInfo);
+                    void connect(connectionInfo);
```

Comment:
Nit: I'd have a preference for this to happen inside the connect action to be honest, even if it's just dispatching the same event based on some configuration option that's getting passed to the method, I don't think we should give the connection form more control of the flow than needed here

Reply:
OK I'll add an options parameter to connect.

Reply:
(I ended up adding separate action creators.)